src/LegacyLinkBlockRepository.php

```diff
@@ -62,11 +62,13 @@ public function __construct(Db $db, Shop $shop, Translator $translator)
     public function getByIdHook($id_hook)
     {
         $id_hook = (int) $id_hook;
+        $shopId = Context::getContext()->shop->id;
 
-        $sql = "SELECT cb.`id_link_block`
-                    FROM {$this->db_prefix}link_block cb
-                    WHERE `id_hook` = $id_hook
-                    ORDER by cb.`position`
+        $sql = "SELECT lb.`id_link_block`
+                    FROM {$this->db_prefix}link_block lb
+                    INNER JOIN {$this->db_prefix}link_block_shop lbs ON lbs.`id_link_block` = lb.`id_link_block`
+                    WHERE lb. `id_hook` = $id_hook AND lbs.`id_shop` = {$shopId}
+                    ORDER by lb.`position`
```

Comment:
Probably needs to be refactored to support sorting per store, huge BC break 👎

Reply:
BC breaks are allowed if we release a new major version of the module. Also they are quite OK if we provide a migration path from previous to new version.